### PR TITLE
Explain the directory structure with composer-installer v3 vers v4@dev

### PR DESCRIPTION
**Pull request by linawolf**
releases: 11.5

### CODE REVIEW

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 1]
```diff
@@ -84,32 +84,10 @@ We assume here that your web root points to a folder called :file:`public` in
...
-    The Composer plugin `typo3/cms-composer-installers` in version 4+ was created
-    for TYPO3 version 12 and backported for default but **optional usage**
-    in TYPO3 version 11. Therefore the version has to be explicitely set (decreased)
-    if the classical directory structure shall be used:
-
-.. code-block:: none
-
-   "typo3/cms-composer-installers": "^3.1",
+..  note::
+    If you find a directory called :file:`_assets` in this directory you are
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:88]:
```suggestion
    If you find a directory called :file:`_assets/` in this directory, you are
```

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 2]
```diff
@@ -84,32 +84,10 @@ We assume here that your web root points to a folder called :file:`public` in
...
+    either running :ref:`composer-installer-v4` or
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:89]:
```suggestion
    running either :ref:`composer-installer-v4` or
```

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 3]
```diff
@@ -148,6 +126,11 @@ each system extension the PHP files are located in the folder
 :file:`Classes/`. See :ref:`extension files locations <extension-files-locations>`
 for more information on how single extensions are structured.
 
+..  note::
+    If you cannot find the TYPO3 Core source in the directory
+    :file:`public/typo3/sysext` you are either running :ref:`composer-installer-v4` or
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:131]:
```suggestion
    :file:`public/typo3/sysext/` you are running either :ref:`composer-installer-v4` or
```

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 4]
```diff
@@ -157,13 +140,18 @@ Amongst others, this directory contains the files :file:`LocalConfiguration.php`
...
 :file:`public/typo3conf/ext/`
 """""""""""""""""""""""""""""
 
-Directory for local TYPO3 extensions. Each subdirectory contains one extension.
+Directory for third-party and custom TYPO3 extensions. Each subdirectory
+contains one extension.
+
+..  note::
+    If you cannot find the directory called :file:`public/typo3conf/ext/` you are
+    either running :ref:`composer-installer-v4` or
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:153]:
```suggestion
    running either :ref:`composer-installer-v4` or
```

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 5]
```diff
@@ -221,6 +209,51 @@ TYPO3 log, the deprecations log and logs generated by extensions.
 :file:`vendor/`
 ---------------
 
+In this directory, which lies outside of
+the webroot, third-party dependencies that are not TYPO3 extensions are
+installed.
+
+If optional :ref:`Composer installers v4 <composer-installer-v4>`
+are used, TYPO3 extensions are also installed here.
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:217]:
```suggestion
is used, TYPO3 extensions are also installed here.
```

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 6]
```diff
@@ -221,6 +209,51 @@ TYPO3 log, the deprecations log and logs generated by extensions.
...
+
+.. _composer-installer-v4:
+
+TYPO3 v11 with Composer installers v4
+=====================================
+
+The Composer plugin `typo3/cms-composer-installers` in version 4+ was created
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:224]: That is not really true, v4 was there before TYPO3 v12 with v5. In fact, v5 requires just PHP 8.1 as v4 can also be used with PHP 7.4 - no other changes. And v4-rc1 was released months before v5.

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 7]
```diff
@@ -221,6 +209,51 @@ TYPO3 log, the deprecations log and logs generated by extensions.
...
+
+Most notably public assets provided by extensions will be available in
+:ref:`directory-public-assets`
+
+.. _directory-public-assets:
+
+:file:`public/_assets/` (Composer installer v4)
+-----------------------------------------------
+
+This directory includes symlinks to resources of extensions, as consequence
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:245]:
```suggestion
This directory includes symlinks to public resources of extensions, as consequence
```

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 8]
```diff
@@ -221,6 +209,51 @@ TYPO3 log, the deprecations log and logs generated by extensions.
...
+of this and further structure changes the folder :file:`typo3conf/ext/` is
+not created or used anymore.
+So all files like CSS, JavaScript, Icons, Fonts, Images, etc. of extensions
+are not linked anymore directly to the extension folders but to the directory
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:249]:
```suggestion
which are stored in the folder :file:`Resources/Public/` are not available anymore directly to the extension folders but linked into the directory
```

**Review by brotkrueml — CHANGES_REQUESTED** [review]

File: Documentation/ApiOverview/DirectoryStructure/Index.rst (hunk 6 above)
Comment (linawolf) [reply Documentation/ApiOverview/DirectoryStructure/Index.rst:224]: I copied the sentence from above and Have no ideas about this as I was not following it. Could you try to improve this section?

**Comment by brotkrueml** [comment]
@linawolf Rewrote the section about cms-composer-installers v4, please have a look. I also approved the PR for now.

**Review by brotkrueml — APPROVED** [review]

**Pull request merged**